src/test/scala/edu/berkeley/ce/rockslicing/BlockVTKSpec.scala

```diff
@@ -0,0 +1,139 @@
+
+class BlockVTKSpec extends FunSuite {
+  val boundingFaces = List(
+    Face((-1.0, 0.0, 0.0), 0.0, phi=0, cohesion=0), // -x = 0
+    Face((1.0, 0.0, 0.0), 1.0, phi=0, cohesion=0),  // x = 1
+    Face((0.0, -1.0, 0.0), 0.0, phi=0, cohesion=0), // -y = 0
+    Face((0.0, 1.0, 0.0), 1.0, phi=0, cohesion=0),  // y = 1
+    Face((0.0, 0.0, -1.0), 0.0, phi=0, cohesion=0), // -z = 0
+    Face((0.0, 0.0, 1.0), 1.0, phi=0, cohesion=0)   // z = 1
+  )
+  val unitCube = Block((0.0, 0.0, 0.0), boundingFaces)
+
+  val face1 = Face((-1.0, 0.0, 0.0), 0.0, phi=0, cohesion=0)
+  val face2 = Face((1.0, 0.0, 0.0), 1.0, phi=0, cohesion=0)
+  val face3 = Face((0.0, -1.0, 0.0), 0.0, phi=0, cohesion=0)
+  val face4 = Face((0.0, 1.0, 0.0), 1.0, phi=0, cohesion=0)
+  val face5 = Face((0.0, 0.0, -1.0), 0.0, phi=0, cohesion=0)
+  val face6 = Face((0.0, 0.0, 1.0), 1.0, phi=0, cohesion=0)
+  val face1Verts = List((0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (0.0, 0.0, 1.0), (0.0, 1.0, 1.0))
+  val face2Verts = List((1.0, 1.0, 1.0), (1.0, 0.0, 1.0), (1.0, 0.0, 0.0), (1.0, 1.0, 0.0))
+  val face3Verts = List((0.0, 0.0, 1.0), (0.0, 0.0, 0.0), (1.0, 0.0, 0.0), (1.0, 0.0, 1.0))
+  val face4Verts = List((0.0, 1.0, 0.0), (0.0, 1.0, 1.0), (1.0, 1.0, 1.0), (1.0, 1.0, 0.0))
+  val face5Verts = List((1.0, 1.0, 0.0), (1.0, 0.0, 0.0), (0.0, 0.0, 0.0), (0.0, 1.0, 0.0))
+  val face6Verts = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0))
+
+  def doubleListElementDiff(list1: Seq[Double], list2: Seq[Double]): Seq[Double] = {
+    assert(list1.length == list2.length)
+    val listDiff  = Seq.empty[Double]
+    for (i <- 0 until list1.length) {
+      listDiff +: Seq[Double](math.abs(list1(i) - list2(i)))
+    }
+    listDiff
+  }
+
+  def intListElementDiff(list1: Seq[Int], list2: Seq[Int]): Seq[Int] = {
+    assert(list1.length == list2.length)
+    val listDiff  = Seq.empty[Int]
+    for (i <- 0 until list1.length) {
+      listDiff +: Seq[Int](math.abs(list1(i) - list2(i)))
+    }
+    listDiff
+  }
+
+  test("The vertices should be oriented counter-clockwise for each face") {
+    val expectedVertices = Map(
+      face1 -> face1Verts,
+      face2 -> face2Verts,
+      face3 -> face3Verts,
+      face4 -> face4Verts,
+      face5 -> face5Verts,
+      face6 -> face6Verts
+    )
+    val vtkBlock = BlockVTK(unitCube)
+    assert(expectedVertices == vtkBlock.orientedVertices)
+  }
+
+  test("List of vertices should contain only distinct vertices as tuples") {
+    val vtkBlock = BlockVTK(unitCube)
+    val expectedVertices = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0),
+                                (0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (1.0, 0.0, 0.0), (1.0, 1.0, 0.0))
+    assert(expectedVertices == vtkBlock.tupleVertices)
+  }
+
+  test("List of tuple vertices should flatten into list of doubles") {
+    val vtkBlock = BlockVTK(unitCube)
+    val expectedVertices = List((0.0, 1.0, 1.0), (0.0, 0.0, 1.0), (1.0, 0.0, 1.0), (1.0, 1.0, 1.0),
```

Comment:
Is there a reason not to just write this as a list of doubles and dispense with the flatMap?